tests/unit/deploy/test_packager.py

```diff
@@ -0,0 +1,253 @@
+        # Test that `pip wheel` is called with the correct params
+        pip, runner = pip_runner
+        whl = 'foobar-1.0-py3-none-any.whl'
+        directory = 'directory'
+        runner.build_wheel(whl, directory)
+        assert pip.calls[0] == ['wheel', '--no-deps', '--wheel-dir',
+                                directory, whl]
+
+    def test_download_all_deps(self, pip_runner):
+        # Make sure that `pip download` is called with the correct arguments
+        # for getting all sdists.
+        pip, runner = pip_runner
+        runner.download_all_dependencies('requirements.txt', 'directory')
+        assert pip.calls[0] == ['download', '-r',
+                                'requirements.txt', '--dest', 'directory']
+
+    def test_download_wheels(self, pip_runner):
+        # Make sure that `pip download` is called with the correct arguments
+        # for getting lambda compatible wheels.
+        pip, runner = pip_runner
+        packages = ['foo', 'bar', 'baz']
+        runner.download_manylinux_whls(packages, 'directory')
+        if sys.version_info[0] == 2:
+            abi = 'cp27mu'
+        else:
+            abi = 'cp36m'
+        expected_prefix = ['download', '--only-binary=:all:', '--no-deps',
+                           '--platform', 'manylinux1_x86_64',
+                           '--implementation', 'cp', '--abi', abi,
+                           '--dest', 'directory']
+        for i, package in enumerate(packages):
+            assert pip.calls[i] == expected_prefix + [package]
+
+    def test_download_wheels_no_wheels(self, pip_runner):
+        pip, runner = pip_runner
+        runner.download_manylinux_whls([], 'directory')
+        assert len(pip.calls) == 0
+
+
+class TestSdistMetadataFetcher(object):
+    _SETUPTOOLS = 'from setuptools import setup'
+    _DISTUTILS = 'from distutils.core import setup'
+    _BOTH = (
+        'try:\n'
+        '    from setuptools import setup\n'
+        'except ImportError:\n'
+        '    from distutils.core import setuptools\n'
+    )
+
+    _SETUP_PY = (
+        '%s\n'
+        'setup(\n'
+        '    name="%s",\n'
+        '    version="%s"\n'
+        ')\n'
+    )
+
+    def _write_fake_sdist(self, setup_py, directory, ext):
+        filename = 'sdist.%s' % ext
+        path = '%s/%s' % (directory, filename)
+        if ext == 'zip':
+            with zipfile.ZipFile(path, 'w',
+                                 compression=zipfile.ZIP_DEFLATED) as z:
+                z.writestr('sdist/setup.py', setup_py)
+        else:
+            with tarfile.open(path, 'w:gz') as tar:
+                tarinfo = tarfile.TarInfo('sdist/setup.py')
+                tarinfo.size = len(setup_py)
+                tar.addfile(tarinfo, io.BytesIO(setup_py.encode()))
+        return directory, filename
+
+    def test_setup_tar_gz(self, osutils, sdist_reader):
+        setup_py = self._SETUP_PY % (
+            self._SETUPTOOLS, 'foo-bar', '1.0-2b'
```

Comment:
And we are using `'1.0-2b'` and  `'foo-bar'` because those won't work with the naive approach? It might be worth making it clear by having an explicit test name for it. And also have a test for a normal package name and version.